codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/specs/test-service-version-user-agent-class.java

```diff
@@ -0,0 +1,15 @@
+package software.amazon.awssdk.services.json.internal;
+
+import java.lang.String;
+import software.amazon.awssdk.annotations.Generated;
+
+@Generated("software.amazon.awssdk:codegen")
+public final class ServiceVersionUserAgent {
+  /**
+   * Returns a user agent containing the service and version info
+   */
+  public static final String USER_AGENT = "Json_Service#{{VERSION}}";
```

Comment:
IIRC, the key should be the service ID

Reply:
By key do you mean the variable name `USER_AGENT` should be the name of the service? Or are you referring to the service identifier before the #? 

The `Json_Service` part is the fixture file equivalent of the service id:
codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/specs/ServiceVersionUserAgentSpec.java

This will result in 

public static final String USER_AGENT = "Dynamodb#2.26.1"
Later on we append the `api` prefix so the useragent is: `api/Dynamodb#2.26.1`